### PR TITLE
Add conflict behavior when batch-inserting ontology types

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

We have a potential data race when inserting external types as a (direct or indirect) dependency either another ontology type or an entity to be inserted. Precisely, this can happen if two operations try to insert the same external dependency at the same time. One insertion will fail with a duplicate key violation.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204000740778938/1203958456055107/f) _(internal)_

## 🔍 What does this change?

- Adds `ConflictBehavior` to batch insertion methods_

## 🚫 Blocked by

- [x] This is currently a POC. The behavior for `ConflictBehavior` still needs to be implemented

### CODE REVIEW

File: apps/hash-graph/tests/integration/postgres/lib.rs [hunk 1]
```diff
@@ -137,7 +137,9 @@ impl DatabaseTestWrapper {
 
             (data_type, metadata)
         });
-        store.create_data_types(data_types_iter).await?;
+        store
+            .create_data_types(data_types_iter, ConflictBehavior::Skip)
```
Comment (thehabbos007) [inline apps/hash-graph/tests/integration/postgres/lib.rs:141]: Was `skip` chosen for the integration tests for a reason? I'm just curious :)

**Review by thehabbos007 — DISMISSED** [review]
Looks good to me, thanks!

File: apps/hash-graph/tests/integration/postgres/lib.rs (hunk 1 above)
Comment (TimDiekmann) [reply apps/hash-graph/tests/integration/postgres/lib.rs:141]: Yes, it was quite annoying that I had to wipe the database every time I tried running the test. The possibility to skip those was quite nice.

Comment (thehabbos007) [reply apps/hash-graph/tests/integration/postgres/lib.rs:141]: That's actually a pretty good reason, and something I've experienced as well. Thanks for explaining!

**Review by thehabbos007 — DISMISSED** [review]

**Review by thehabbos007 — APPROVED** [review]

**Pull request merged**